kasa/smart/modules/lightstripeffect.py

```diff
@@ -0,0 +1,109 @@
+"""Module for light effects."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from ...interfaces.lighteffect import LightEffect as LightEffectInterface
+from ..effects import EFFECT_MAPPING, EFFECT_NAMES
+from ..smartmodule import SmartModule
+
+if TYPE_CHECKING:
+    pass
+
+
+class LightStripEffect(SmartModule, LightEffectInterface):
+    """Implementation of dynamic light effects."""
+
+    REQUIRED_COMPONENT = "light_strip_lighting_effect"
+
+    @property
+    def name(self) -> str:
+        """Name of the module."""
+        return "LightEffect"
+
+    @property
+    def effect(self) -> str:
+        """Return effect state.
+
+        Example:
+            {'brightness': 50,
+             'custom': 0,
+             'enable': 0,
+             'id': '',
+             'name': ''}
+        """
+        if (
+            (lighting_effect := self.data.get("lighting_effect"))
+            and lighting_effect.get("enable")
+            and (name := lighting_effect.get("name"))
+            and name in EFFECT_NAMES
+        ):
```

Comment:
I'd split this up for readability.

Reply:
So I was thinking about something like this:
```
eff = self.data["lighting_effect"]
name = eff["name"]
if eff["enable"]:
    return name

return self.LIGHT_EFFECTS_OFF
```
or just create a class inheriting from pydantic.BaseModel, so it'll explode if we receive unexpected data. Having a stack trace is less annoying than trying to figure out why the library is reporting wrong information (in this case, if the reported name of the effect is not in our list, it'll say that there's no effect active).

Reply:
I went with the suggestion above.  Currently avoiding pydantic in the hope we move away from it :smile: